tests/foreman/api/test_remoteexecution.py

```diff
@@ -72,3 +75,137 @@ def test_positive_run_capsule_upgrade_playbook(module_capsule_configured, target
     ).refresh()
     feature_set = {feat['name'] for feat in result['features']}
     assert {'Ansible', 'Dynflow', 'Script', 'Pulpcore', 'Logs'}.issubset(feature_set)
+
+
+@pytest.mark.tier3
+@pytest.mark.no_containers
+@pytest.mark.rhel_ver_match('[^6].*')
+def test_negative_time_to_pickup(
+    module_org,
+    module_target_sat,
+    smart_proxy_location,
+    module_ak_with_cv,
+    module_capsule_configured_mqtt,
+    rhel_contenthost,
+):
+    """Time to pickup setting is honored for host registered to mqtt
+
+    :id: a082f599-fbf7-4779-aa18-5139e2bce774
+
+    :expectedresults: Time to pickup aborts the job if mqtt client doesn't
+        respond in time
+
+    :CaseImportance: High
+
+    :bz: 2158738, 2118651
+
+    :parametrized: yes
+    """
+    client_repo = ohsnap.dogfood_repository(
+        settings.repos.ohsnap_repo_host,
+        product='client',
+        repo='client',
+        release='Client',
+        os_release=rhel_contenthost.os_version.major,
+    )
+    # Update module_capsule_configured_mqtt to include module_org/smart_proxy_location
+    module_target_sat.cli.Capsule.update(
+        {
+            'name': module_capsule_configured_mqtt.hostname,
+            'organization-ids': module_org.id,
+            'location-ids': smart_proxy_location.id,
+        }
+    )
+    # register host with pull provider rex
+    result = rhel_contenthost.register(
+        module_org,
+        smart_proxy_location,
+        module_ak_with_cv.name,
+        target=module_capsule_configured_mqtt,
+        satellite=module_target_sat,
+        setup_remote_execution_pull=True,
+        repo=client_repo.baseurl,
+    )
+    template_id = (
+        module_target_sat.api.JobTemplate()
+        .search(query={'search': 'name="Run Command - Script Default"'})[0]
+        .id
+    )
+    assert result.status == 0, f'Failed to register host: {result.stderr}'
+    # check mqtt client is running
+    result = rhel_contenthost.execute('yggdrasil status')
+    assert result.status == 0, f'Failed to start yggdrasil on client: {result.stderr}'
```

Comment:
Just beware this is not as great indication as one could think
```
# yggdrasil status
Connection status for joint-jaybird.example.com:

● Connected to Red Hat Subscription Management
● The yggdrasil daemon is inactive

Manage your Red Hat connector systems: https://red.ht/connector

# echo $?
0

# systemctl status yggdrasild
● yggdrasild.service - yggdrasil daemon
   Loaded: loaded (/usr/lib/systemd/system/yggdrasild.service; enabled; vendor preset: disabled)
   Active: failed (Result: exit-code) since Thu 2023-02-02 08:40:22 EST; 1min 9s ago
     Docs: https://github.com/redhatinsights/yggdrasil
  Process: 849 ExecStart=/usr/sbin/yggdrasild (code=exited, status=1/FAILURE)
 Main PID: 849 (code=exited, status=1/FAILURE)

Feb 02 08:40:22 joint-jaybird.example.com systemd[1]: Started yggdrasil daemon.
Feb 02 08:40:22 joint-jaybird.example.com yggdrasild[849]: [yggdrasild] 2023/02/02 08:40:22 /builddir/build/BUILD/yggdrasil-0.2.0/cmd/yggd/main.go:155: starting yggdrasild version 0.2.0
Feb 02 08:40:22 joint-jaybird.example.com yggdrasild[849]: [yggdrasild] 2023/02/02 08:40:22 /builddir/build/BUILD/yggdrasil-0.2.0/cmd/yggd/main.go:200: listening on socket: @yggd-dispatcher-oIHvYx
Feb 02 08:40:22 joint-jaybird.example.com yggdrasild[849]: cannot connect to broker: network Error : dial tcp: lookup expert-earwig.example.com on 192.168.122.1:53: no such host
Feb 02 08:40:22 joint-jaybird.example.com systemd[1]: yggdrasild.service: Main process exited, code=exited, status=1/FAILURE
Feb 02 08:40:22 joint-jaybird.example.com systemd[1]: yggdrasild.service: Failed with result 'exit-code'.
```